qoajad_backend/src/main/java/com/qoajad/backend/database/DatabaseAccessorImplementation.java

```diff
@@ -0,0 +1,143 @@
+package com.qoajad.backend.database;
+
+import com.qoajad.backend.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@Qualifier(value = "defaultDatabaseAccessor")
+public class DatabaseAccessorImplementation implements DatabaseAccessor {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    // TODO Create exception class to handle every catch on the CRUD functions in order to avoid duplicate code
+
+    @Autowired
+    public DatabaseAccessorImplementation(final JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+        Objects.requireNonNull(jdbcTemplate, "To instantiate the database accessor the jdbc template must not be null.");
+    }
+
+    @Override
+    public ResponseEntity<List<User>> retrieveAllUsers() {
+        final String query = "SELECT user_id, user_pw FROM User";
+        List<User> users;
+        ResponseEntity<List<User>> response;
+        try {
+            users = jdbcTemplate.query(query, (resultSet, rowNum) -> {
+                final int userId = resultSet.getInt("user_id");
+                final String password = resultSet.getString("user_pw");
+                return new User(userId, password);
+            });
+            response = new ResponseEntity<>(users, HttpStatus.OK);
+        } catch (Exception e) {
+            if(e instanceof EmptyResultDataAccessException) {
```

Comment:
I don't think is a good idea to use instanceof for exceptions since you don't know the hierarchy that the exceptions use and the instanceof operator considers that. For example, 

Class Animal {
}
Class Dog extends Animal {
}

final Dog dog = new Dog();
dog instanceof Animal // Returns true
Animal instanceof Animal // Returns true

And since you are using else ifs, perhaps the first if is a more general if, and you wanted your exception to rely on the second if (Which will not happen) So you won't be able to recover from these exceptions correctly and as you can see this is very error prone. Instead catch multiple exceptions.